src/proxy.sol

```diff
@@ -98,10 +118,22 @@ contract ProxyRegistry is Title {
     // deploys a new proxy instance
     // sets custom owner of proxy by issuing an Title NFT
     function build(address owner) public returns (address payable proxy) {
-        uint token = _issue(owner);
-        proxy = address(new Proxy(token));
-        proxies[token] = proxy;
-        emit Created(msg.sender, owner, proxy, token);
+        uint accessToken = _issue(owner);
+        bytes32 salt = keccak256(abi.encodePacked(accessToken));
+        proxy = deploy(accessToken, salt);
+        emit Created(msg.sender, owner, proxy, accessToken);
+    }
+
+
+    /// uses the create2 opcode to deploy a proxy contract
+    function deploy(uint accessToken, bytes32 salt) internal returns (address payable addr) {
```

Comment:
If deploy is only called from the build method, why not merge the two functions?